packages/forms-web-app/src/dynamic-forms/controller.js

```diff
@@ -238,23 +242,44 @@ exports.save = async (req, res, journeyId) => {
 		return res.render(`dynamic-components/${questionObj.type}/index`, viewModel);
 	}
 
-	// use custom saveAction
-	if (questionObj.saveAction) {
-		return await questionObj.saveAction(req, res);
-	}
+	try {
+		// use custom saveAction
+		if (questionObj.saveAction) {
+			return await questionObj.saveAction(req, res);
+		}
+
+		// set answer on response
 
-	// set answer on response
-	journeyResponse.answers[questionObj.fieldName] = req.body[questionObj.fieldName];
-	for (let propName in req.body) {
-		if (propName.startsWith(questionObj.fieldName + '_')) {
-			journeyResponse.answers[propName] = req.body[propName];
+		journeyResponse.answers[questionObj.fieldName] = req.body[questionObj.fieldName];
+		for (let propName in req.body) {
+			if (propName.startsWith(questionObj.fieldName + '_')) {
+				journeyResponse.answers[propName] = req.body[propName];
+			}
 		}
-	}
 
-	// save response to session
-	saveResponseToSessionByType(req, journeyResponse);
+		// save answer to database
+		const answerToSave = { [questionObj.fieldName]: req.body[questionObj.fieldName] };
```

Comment:
Looking at this in conjunction with the section above, there may be more than one answer to save. In the section above, any field prefixed with the field name is also persisted in the version of the response stored in the session, but the patch sent to the DB only updates the field whose name exactly matches the specified fieldname. We'll need those other fields persisted to allow us to handle things like address collection and checkboxes with a conditional field (e.g. radio buttons with an "Other - please specify" option which shows a text field).